zitadel/terraform/main.tf

```diff
@@ -0,0 +1,95 @@
+terraform {
+  required_providers {
+    zitadel = {
+      source  = "zitadel/zitadel"
+      version = "1.0.0-alpha.18"
+    }
+  }
+}
+
+provider "zitadel" {
+  domain           = "iam.xpanse.site"
+  insecure         = "false"
+  port             = "443"
+  jwt_profile_file = "local-token.json"
+}
+
+resource "zitadel_org" "eclipse-xpanse" {
+  name = "eclipse-xpanse"
+}
+
+resource zitadel_project eclipse-xpanse {
+  name                     = "eclipse-xpanse"
+  org_id                   = zitadel_org.eclipse-xpanse.id
+  project_role_assertion   = true
+  project_role_check       = false
+  has_project_check        = false
+  private_labeling_setting = "PRIVATE_LABELING_SETTING_ENFORCE_PROJECT_RESOURCE_OWNER_POLICY"
+}
+
+resource zitadel_application_api application_api {
+  org_id           = zitadel_org.eclipse-xpanse.id
+  project_id       = zitadel_project.eclipse-xpanse.id
+  name             = "xpanse-api"
+  auth_method_type = "API_AUTH_METHOD_TYPE_PRIVATE_KEY_JWT"
+}
+
+resource zitadel_application_oidc xpanse-ui {
+  project_id = zitadel_project.eclipse-xpanse.id
+  org_id     = zitadel_org.eclipse-xpanse.id
+
+  name                        = "xpanse-ui"
+  redirect_uris               = ["https://localhost:3000"]
+  response_types              = ["OIDC_RESPONSE_TYPE_CODE"]
+  grant_types                 = ["OIDC_GRANT_TYPE_AUTHORIZATION_CODE"]
+  app_type                    = "OIDC_APP_TYPE_USER_AGENT"
+  auth_method_type            = "OIDC_AUTH_METHOD_TYPE_NONE"
+  version                     = "OIDC_VERSION_1_0"
+  clock_skew                  = "0s"
+  dev_mode                    = true
+  access_token_type           = "OIDC_TOKEN_TYPE_BEARER"
+  access_token_role_assertion = false
+  id_token_role_assertion     = true
+  id_token_userinfo_assertion = true
+  additional_origins          = []
+}
+
+resource zitadel_application_oidc swagger-ui {
+  project_id = zitadel_project.eclipse-xpanse.id
+  org_id     = zitadel_org.eclipse-xpanse.id
+  name                        = "swagger-ui"
+  redirect_uris               = ["http://localhost:8080/swagger-ui/oauth2-redirect.html"]
```

Comment:
we must move this value to a variables.tf file

Reply:
fixed